validator/static/validator/js/metadata_consumer.js

```diff
@@ -0,0 +1,123 @@
+                file_name: file.name,
+                dirHandle: dirHandle
+            }
+            const { results, error } = await asyncRun(validate_metadata, context);
+            if(results){
+                console.log(results);
+                spinner.style.visibility = "hidden";
+                showResults(results);
+            }
+            if(error){
+                console.error(error);
+                spinner.style.visibility = "hidden";
+                showSystemError(error);
+            }
+        };
+        reader.readAsArrayBuffer(file);
+    }
+}
+
+function report_items_2_html(reports_list) {
+    let report = '<ul>';
+    $.each(reports_list, function(index, report_item){
+      let lines = ''
+      if (report_item.lines) {
+        let lines_label = (report_item.lines.length > 1) ? 'Lines' : 'Line';
+        lines = lines_label+": "+report_item.lines.join(',')+ ' &rarr; ';
+      }
+      let message = report_item.message;
+      // Value highlighting
+      message = message.replace(/"(.+?)"/g, "\"<b>$1</b>\"");
+      // Leading space
+      message = message.replace(/"<b>\s+/g, "\"<b><span class=\"pgs_color_red\">_</span>");
+      // Trailing space
+      message = message.replace(/\s+<\/b>"/g, "<span class=\"pgs_color_red\">_</span></b>\"");
+      // Column highlighting
+      message = message.replace(/'(.+?)'/g, "\'<span class=\"pgs_color_1\">$1</span>\'");
+      report += "<li>"+lines+message+"</li>";
+    });
+    report += '</ul>';
+    return report;
+}
+
+function makeReportTable(data_spreadsheet_items, items_header){
+    let table_html = '<table class="table table-bordered" style="width:auto"><thead class="thead-light">'+
+                     '<tr><th>Spreadsheet</th><th>'+items_header+'</th></tr>'+
+                     '</thead><tbody>';
+        $.each(data_spreadsheet_items, function(spreadsheet, reports_list){
+          table_html += "<tr><td><b>"+spreadsheet+"</b></td><td>";
+          table_html += report_items_2_html(reports_list);
+          table_html += '</td></tr>';
+        });
+        table_html += '</tbody></table>';
+        return table_html;
+}
+
+function showResults(results){
+    let data = JSON.parse(results);
+    let status_style = (data.status === 'failed') ? '<i class="fa fa-times-circle pgs_color_red" style="font-size:18px"></i> Failed' : '<i class="fa fa-check-circle pgs_color_green" style="font-size:18px"></i> Passed';
+    let status_html = '<table class="table table-bordered table_pgs_h mb-4"><tbody>'+
+                        '  <tr><td>File validation</td><td>'+status_style+'</td></tr>'+
+                        '</tbody></table>';
+    $('#check_status').html(status_html);
+    // Error messages
+    if (data.error) {
+        let report = '<h5 class="mt-4"><i class="fa fa-times-circle pgs_color_red"></i> Error report</h5>'
+            + makeReportTable(data.error, 'Error message(s)');
+        $('#report_error').html(report);
```

Comment:
Might be overkilling, but wondering if we should have a function for these `html reports`, with the type of report and the message as arguments, e.g.:
```
function reports_html(type,msg) {
    $('#report_'+type).html(msg);
}
```

Reply:
I feel like this refactoring would make the code a bit confusing